### PR TITLE
feat(shared): attachment improvements

### PR DESCRIPTION
**Pull request by ragi96**
Small UX Improvments for the attachments component 

They are coming from the review of the audio block [task](https://app.clickup.com/t/ndbnjd)

### CODE REVIEW

File: packages/shared/src/components/Attachments/Attachments.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ import {
 import { useGuidelineDesignTokens } from '../../hooks';
 import { AttachmentItem, SortableAttachmentItem } from './AttachmentItem';
 import { AttachmentsProps } from './types';
+import { Tooltip, TooltipPosition } from '@frontify/fondue';
 
 export const Attachments = ({
     items,
```
[the remark this answers is not in the document]
Comment (triggertoo) [reply packages/shared/src/components/Attachments/Attachments.tsx:30]: also:
`const [internalItems, setInternalItems] = useState<Asset[]>(items);`

File: packages/shared/src/components/Attachments/Attachments.tsx [hunk 2]
```diff
@@ -120,84 +126,92 @@ export const Attachments = ({
...
+                                    {isEditing && (
+                                        <div className="tw-px-5 tw-py-3">
+                                            <div className="tw-font-body tw-font-medium tw-text-text tw-text-s tw-my-4">
+                                                Add attachments
+                                            </div>
+                                            <AssetInput
+                                                isLoading={isUploadLoading}
+                                                onMultiAssetClick={() => {
+                                                    /* This enables multiple file inputs */
+                                                }}
```
Comment (triggertoo) [inline packages/shared/src/components/Attachments/Attachments.tsx:203]: can be removed

File: packages/shared/src/components/Attachments/Attachments.tsx [hunk 3]
```diff
@@ -120,84 +126,92 @@ export const Attachments = ({
...
+                            <Flyout
+                                placement={FlyoutPlacement.BottomRight}
+                                onOpenChange={setIsFlyoutOpen}
+                                isOpen={isFlyoutOpen}
+                                fitContent
+                                legacyFooter={false}
+                                trigger={
+                                    <button className="tw-flex tw-text-[13px] tw-font-body tw-items-center tw-gap-1 tw-rounded-full tw-bg-box-neutral-strong-inverse hover:tw-bg-box-neutral-strong-inverse-hover active:tw-bg-box-neutral-strong-inverse-pressed  tw-text-box-neutral-strong tw-outline tw-outline-1 tw-outline-offset-[1px] tw-p-[6px] tw-outline-line">
+                                        <IconPaperclip16 />
+                                        <div>{(items?.length || 0) > 0 ? items?.length : 'Add'}</div>
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/shared/src/components/Attachments/Attachments.tsx]: adjusted

File: packages/shared/src/components/Attachments/Attachments.tsx (hunk 2 above)
Comment (ragi96) [reply packages/shared/src/components/Attachments/Attachments.tsx]: removed

File: packages/shared/src/components/Attachments/Attachments.tsx [hunk 4]
```diff
@@ -120,84 +126,92 @@ export const Attachments = ({
...
+                                        <IconCaretDown12 />
+                                    </button>
+                                }
+                            >
+                                <div className="tw-w-[300px]">
+                                    {(internalItems?.length || 0) > 0 && (
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/shared/src/components/Attachments/Attachments.tsx]: adjusted

File: packages/shared/src/components/Attachments/Attachments.tsx [hunk 5]
```diff
@@ -120,84 +126,92 @@ export const Attachments = ({
...
+                                        <DndContext
+                                            sensors={sensors}
+                                            collisionDetection={closestCenter}
+                                            onDragStart={handleDragStart}
+                                            onDragEnd={handleDragEnd}
+                                        >
+                                            <SortableContext items={internalItems || []} strategy={rectSortingStrategy}>
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/shared/src/components/Attachments/Attachments.tsx]: adjusted

File: packages/shared/src/components/Attachments/Attachments.tsx [hunk 6]
```diff
@@ -120,84 +126,92 @@ export const Attachments = ({
...
+                                                <div className="tw-border-b tw-border-b-line">
+                                                    {(internalItems || []).map((item) => (
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/shared/src/components/Attachments/Attachments.tsx]: adjusted

**Review by triggertoo — APPROVED** [review]

**Review by celine-s — APPROVED** [review]
looks good to me 👍

**Pull request merged**